src/protocols/secure_channel/CASESession.cpp

```diff
@@ -343,37 +360,53 @@ CHIP_ERROR CASESession::SendSigmaR1()
         ReturnErrorCodeIf(mFabricInfo == nullptr, CHIP_ERROR_INCORRECT_STATE);
         memcpy(mIPK, GetIPKList()->data(), sizeof(mIPK));
         ReturnErrorOnFailure(
-            mFabricInfo->GenerateDestinationID(ByteSpan(mIPK), ByteSpan(initiatorRandom), GetPeerNodeId(), destinationIdSpan));
+            mFabricInfo->GenerateDestinationID(ByteSpan(mIPK), ByteSpan(mInitiatorRandom), GetPeerNodeId(), destinationIdSpan));
     }
     ReturnErrorOnFailure(tlvWriter.PutBytes(TLV::ContextTag(3), destinationIdentifier, sizeof(destinationIdentifier)));
 
     ReturnErrorOnFailure(
         tlvWriter.PutBytes(TLV::ContextTag(4), mEphemeralKey.Pubkey(), static_cast<uint32_t>(mEphemeralKey.Pubkey().Length())));
+
+    // If CASE session was previously established using the current state information, let's fill in the session resumption
+    // information in the the Sigma1 request. It'll speed up the session establishment process if the peer can resume the old
+    // session.
+    if (mPairingComplete)
+    {
+        ReturnErrorOnFailure(tlvWriter.PutBytes(TLV::ContextTag(6), mResumptionId, kCASEResumptionIDSize));
+
+        chip::Platform::ScopedMemoryBuffer<uint8_t> initiatorResume1MIC;
```

Comment:
Please use a stack buffer for `initiatorResume1MIC`. Resume1MIC is 16 bytes. The stack usage of calling `Alloc` here goes deeper than that.